src/classes/ProfileResults.ts

```diff
@@ -13,46 +31,19 @@ module.exports = class ProfileResults {
    *
    * It is also helpful for providing useful jsdoc
    * definitions for the properties of the data.
-   *
-   * @param {Object} data
-   *
-   * @param {Number} data.start
-   * When the profiling started.
-   *
-   * @param {Number} data.end
-   * When the profiling was stopped.
-   *
-   * @param {Number} data.time_elapsed
-   * The amount of time the profiler was active, in milliseconds.
-   *
-   * @param {Number} data.timestep_ms
-   * The amount of milliseconds that would pass before each memory check.
-   *
-   * @param {Array} data.mem_list
-   * The list of memory values throughout the test.
-   *
-   * @param {Number} data.start_usage_bytes
-   * The amount of bytes that the process took up when profiler started.
-   *
-   * @param {Number} data.peak_usage_bytes
-   * The highest amount of memory usage.
-   *
-   * @param {Number} data.end_usage_bytes
-   * The amount of bytes the process took up when the profile was finished.
-   *
-   * @param {Number} data.base_process_bytes
-   * The amount of bytes the process took up without anything happening.
    */
-  constructor (data) {
+  constructor(data: ResultData | null) {
+    if (!data) throw Error('Results data was null');
```

Comment:
Might be just me but I'm not a fan of inline if's. Keep doing it if you want though, it's mainly opinion.